### DISCUSSION TITLE
Why would this code route.js seems to be working from my localhost, but not when I visit the link from the live site?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#69056] [Help] [answered]
### Summary

Problem:
I got an href link pointing at a route.js page.  Testing the link from the localhost works fine, but when I test the live site is doesn't work. 

href looks like this:

```
<a href="https://www.email.tu.biz/my/social/facebook/${email}">
                  <img
                    src="https://email.tu.biz/_next/image?url=%2Fimages%2Fsocial-icons%2Ffacebook24.png&w=1920&q=75"
                    alt="Facebook"
                    style="
                      max-width: 100%;
                      height: auto;
                      margin-left: auto;
                      margin-right: auto;
                    "
                  />
                </a>
```

^^^ This DOESN'T  work.

`http://localhost:3000/my/social/facebook/email`

^^^ This DOES work.  It sends me an email does the proper re-direct when testing for Facebook. (I haven't tested the other links yet)

`https://www.email.tu.biz/my/social/facebook/email`

^^^ This DOESN'T work when I type the address directly on the browser.

The route.js code is this:

```
import nodemailer from "nodemailer";
import { NextResponse } from "next/server";

export async function GET(request, { params }) {
  // Assuming `params` is used to extract `linkClick`
  const linkClick = params.link;
  const userEmail = params.email;

  const transporter = nodemailer.createTransport({
    service: "Outlook365",
    auth: {
      user: process.env.EMAIL2,
      pass: process.env.EPASSWORD2,
    },
  });

  const mailData = {
    from: { name: "Edgar Lindo1", address: process.env.EMAIL2 },
    to: "edgar@tu.biz",
    subject: `We got feedback from ${linkClick}`,
    text: ``,
    html: `<div>
            <div>Hello There,</div>
            <div>This is a click. ${linkClick}</div>
            <div>He clicked on ${userEmail}</div>
          </div>`,
  };

  try {
    await transporter.sendMail(mailData);
    console.log("email sent");
  } catch (err) {
    console.error("Failed to send email:", err);
    // You might want to handle the error, return an error response, etc.
    return NextResponse.error();
  }

  // Ensure you use `request.url` if `request` is used
  const redirectUrl = new URL("https://www.youtube.com", request.url);

  switch (linkClick) {
    case "facebook":
      redirectUrl.href = "https://www.facebook.com";
      break;
    case "instagram":
      redirectUrl.href = "https://www.instagram.com";
      break;
    case "linkedin":
      redirectUrl.href = "https://www.linkedin.com";
      break;
    case "snapchat":
      redirectUrl.href = "https://www.snapchat.com";
      break;
    case "tiktok":
      redirectUrl.href = "https://www.tiktok.com";
      break;
  }

  return NextResponse.redirect(redirectUrl);
}

```

The route.js doing the re-direct is this:
https://github.com/Elindo586/estuff/blob/main/app/my/social/%5Blink%5D/%5Bemail%5D/route.js

This other route is where I have a link in line 431 pointing the the above route
https://github.com/Elindo586/estuff/blob/main/app/my/email-test/route.js

^^ Again, when testing the link as in localhost, the pointed route works fine, but when I replace localhost by the actual website, it does nothing.

**Comment by Elindo586** [comment]
Could this be a firewall problem for this route.js? 

![image](https://github.com/user-attachments/assets/2bca90fc-be43-49ac-bc65-e7233c3b4152)

**Comment by Elindo586** [comment]
Apparently, adding this line of code makes the code execute and it works.  

`export const dynamic = "force-dynamic";`

I got the suggestion to include this other code too, but I deleted and it doesn't seem to need it:

`export const maxDuration = 10;`


^^  I don't think the problem was taking too long to execute the code.  Seems like the problem was with the dynamic routes not working.   This is implemented in a route.js which execute an email sent, and at the end of the code does a re-direct.

It seems almost silly that I have to `force-hydamic `to make it work, why not make it standard? 


I have a different piece of code not working, when I was trying to do a workaround on this problem.   The force-dynamic didn't work there, maybe because the actual logics to send an email was handle from another route, and the page was just sending a fetch request and a redirect.....  but I am not going to get into why this other code doesn't work because it is just a workaround that I don't need now.

**Comment by Elindo586** [comment, marked as answer]
OK...  I figured out the real reason why it wasn't working.

It was the `href=https://www.mywebsite`

instead I should of wrote:  `href=https://mywebsite`

Must be something at the settings of the site in Vercel, or something from Godaddy setting... not sure....

I found it by accident.

**Reply by icyJoseph** [reply]
If that's the case, you might want to look into forwarding `www.mydomain` to `mydomain` -
